R/aws_s3.R

```diff
@@ -9,17 +9,25 @@ getS3Folders <- function(bucket, prefix = NULL, ...) {
   limit <- 1000
   query <- list(prefix = prefix, delimiter = "/", "max-keys" = limit, marker = NULL)
   result <- aws.s3::s3HTTP(verb = "GET", bucket = bucket, query = query, parse_response = TRUE, ...)
+  shouldStop <- FALSE
+  nextMarker <- ""
   # Handle pagination for large result set.
-  while (result[["IsTruncated"]] == "true") { # if IsTruncted is true, need to send another request to get the remaining result.
+  while (result[["IsTruncated"]] == "true" & !shouldStop) { # if IsTruncted is true, need to send another request to get the remaining result.
```

Comment:
Leave comment that isTruncated can be true even if there is nothing more to retrieve. (Most likely when querying an empty bucket.)